src/InvoiceGenerator/InvoiceGenerator.BusinessLogic/ValidationBLogic.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Net.Mail;
+
+namespace InvoiceGenerator.BusinessLogic
+{
+    public class ValidationBLogic
+    {
+        public string[] validateUserDetails(string clientName, string clientAddress, string contactName, string contactEmail) // Validate all user inputs on client details entry screen
+        {
+            string clientNameErrorMsg = "No error!", clientAddressErrorMsg = "No error!", contactNameErrorMsg = "No error!", contactEmailErrorMsg = "No error!";
```

Comment:
If you know about them, you might consider using a [Dictionary](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2?view=net-6.0) or [KeyValuePair](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.keyvaluepair?view=net-6.0) rather than a list of strings.

These will allow you to have a single object which can be directly addressed on the other side of the validation call.

You could have something like

```csharp
var validationErrors = Dictionary<string, string>();

if (string.IsNullOrWhiteSpace(clientName))
{
  validationErrors.Add("clientName", "Client Name must not be empty");
}
```

then you could use something like:

```csharp
foreach(KeyValuePair<string, string> entry in validationErrors)
{
  // entry.Key is "clientName"
  // entry.Value = "Client Name must not be empty"
}
// we'll also never enter the foreach if there are no validation errors to deal with
```

Not something to do now. but could be something to think about